### PR TITLE
Add a recipe for useful authoring-focused accessibility tools

### PR DESCRIPTION
**Pull request by thegbomb**
To add some functionality to assist with accessibility I am adding some modules to starshot and making some basic settings changes to improve things.

This is a WIP to share code and progress things to a stable state

### CODE REVIEW

File: recipes/starshot_accessibility/recipe.yml
Comment (phenaproxima) [inline recipes/starshot_accessibility/recipe.yml:16]: I'd actually suggest moving these Gin settings directly into `starshot_admin_theme`, and dropping it from the `recipes` list here. If these are better for accessibility, I see no reason for them not to be the defaults.

File: recipes/starshot_accessibility/recipe.yml
Comment (phenaproxima) [inline recipes/starshot_accessibility/recipe.yml:7]: Not sure we should add this. It's only got 242 installs, and it's not clear from the project page what exactly it _does_, beyond a vague "makes forms more accessible". I think this is something we should consider adding separately (i.e., new issue).

File: recipes/starshot_accessibility/recipe.yml
Comment (phenaproxima) [inline recipes/starshot_accessibility/recipe.yml:9]: Same here -- there's a separate issue already for adding this module (#82), so if it's not strictly necessary for this recipe, maybe it'd be best to remove it.

File: recipes/starshot_accessibility/recipe.yml
Comment (thegbomb) [reply recipes/starshot_accessibility/recipe.yml]: The block config yml file is in recipes/starshot/config and just importing on installation, but I'll remove the tweak here adding a role of complimentary was more about showing off that it can be done, rather than it needing to be done. I'll double-check on the value of it and add it back if it is a more solid requirement.

File: recipes/starshot_accessibility/recipe.yml
Comment (thegbomb) [reply recipes/starshot_accessibility/recipe.yml]: Moved to admin_theme recipe

File: recipes/starshot_accessibility/recipe.yml
Comment (thegbomb) [reply recipes/starshot_accessibility/recipe.yml]: I was using a11y_form_helpers as it came from the recommendation on this page:
https://www.drupal.org/docs/getting-started/accessibility/contributed-modules-for-extending-accessibility-in-drupal#s-core-improvements
I only actually used it to ensure inline_form_errors was switched on, which I have now included in the starshot recipe itself.

File: recipes/starshot_accessibility/recipe.yml
Comment (thegbomb) [reply recipes/starshot_accessibility/recipe.yml]: I added a new recipe for starshot_editoria11y but left it as optional like the multilingual. I can remove it and add this in another MR if necessary.

**Comment by thegbomb** [comment]
I have removed other changes which can be dealt with elsewhere, but have left just the editoria11y module being added in this PR.
TBD what happens to it elsewhere.
I have added it to project_browser, but it will not get installed by default.

File: README.md
Comment (phenaproxima) [inline README.md:68]: Let's be a little more "correct" about this.

```suggestion
* [Editoria11y Accessibility Checker](https://www.drupal.org/project/editoria11y)
```

File: recipes/starshot_editoria11y/recipe.yml
Comment (phenaproxima) [inline recipes/starshot_editoria11y/recipe.yml:1]: I think that we want a little bit more generic of a name here. I propose `starshot_accessibility_tools`  as the machine name instead, to match the proposed track name in https://www.drupal.org/project/starshot/issues/3454529. It also gives us a little wiggle room to add _other_ tools here that aren't Editoria11y. (And the human-readable name should be "Accessibility tools" or similar.)

File: recipes/starshot_editoria11y/recipe.yml
Comment (phenaproxima) [inline recipes/starshot_editoria11y/recipe.yml:3]: This should be a little more generic.

```suggestion
description: Installs and configures tools to help authors make their content accessible.
```

File: recipes/starshot_editoria11y/recipe.yml
Comment (phenaproxima) [inline recipes/starshot_editoria11y/recipe.yml:10]: This could use a comment to explain why this element is ignored.

**Review by phenaproxima — CHANGES_REQUESTED** [review]
I think this looks pretty good but mostly could benefit from a rename.

File: recipes/starshot_editoria11y/recipe.yml
Comment (phenaproxima) [inline recipes/starshot_editoria11y/recipe.yml:5]: If `editoria11y` exposes any permissions, we probably want to update the main `starshot` recipe to ensure those permissions are added to the relevant roles (probably any role that can create or edit content should be able to use it).

Comment (thegbomb) [reply recipes/starshot_editoria11y/recipe.yml:5]: It looks like the default config for editoria11y permissions is to grant "View Editoria11y checker" and "Mark as hidden in Editoria11y"  to the authenticated user, which then includes the Content editor role. This could be okay for the site at the moment unless there is a plan to allow non-editors to create accounts.
The other permissions exposed are only granted to the administrator role.
Will recipes allow us to remove permission from a role? If so, I could remove the permissions from the Authenticated role and add it to Content editor.

Comment (phenaproxima) [reply recipes/starshot_editoria11y/recipe.yml:5]: > It looks like the default config for editoria11y permissions is to grant "View Editoria11y checker" and "Mark as hidden in Editoria11y" to the authenticated user

If that's the case, then that's probably okay for now, but can you link me to the spot in Editoria11y's code where it makes this decision?

Comment (thegbomb) [reply recipes/starshot_editoria11y/recipe.yml:5]: From what I can see, it looks like it is set here:
https://git.drupalcode.org/project/editoria11y/-/blob/2.1.x/editoria11y.install?ref_type=heads#L20

**Comment by phenaproxima** [comment]
One typo suggestion, but otherwise I'm good with this. Nice work. I would like to consult with @timplunkett before merging just so that I can be sure that landing this will align with the defined work tracks for Starshot, and grease the wheels for further contribution.

**Comment by phenaproxima** [comment]
I personally have no objection to this, the recipe looks good and ready to me, but I think @pameeela and @timplunkett (from the leadership team) are unsure about adding additional modules to the prototype, even if they are likely to be in the final product.

I'm assigning this to @pameeela to make the final call.

**Comment by pameeela** [comment]
Happy to merge so folks can play around with it easily, I think it's clear that something being in the prototype doesn't imply it will be in the product and @thegbomb is across this. In this case, it's not just adding a module for the sake of it, it's actually providing something that it could be good to get feedback on.

**Pull request merged**
